### PR TITLE
Authentication via SFSafariViewController (iOS 9) and TMWebViewController (up to iOS 8)

### PR DESCRIPTION
**Pull request by shpuntov**
Implemented method `+- (void)authenticate:(NSString *)URLScheme fromViewController:(UIViewController *)fromViewController callback:(void(^)(NSError *))error;` that allows to authenticate on iOS. For iOS up to 8 `TMWebViewController` is used (included in SDK, without XIB, code-only). From iOS 9, `SFSafariViewController` is used. Some minor refactoring included as well (`TMSDKConstants.h` contains a special error domain and error codes for the SDK).

### CODE REVIEW

**Comment by metasmile** [comment]
thanks

**Comment by brianmichel** [comment]
Thanks for the request @shpuntov we'll get to looking at this soon!

**Comment by Pearapps** [comment]
I worry about using a web view as a backup plan here because people could inject javascript and steal users passwords, should we just only allow it on iOS 9?

File: TMTumblrSDK/Authentication/TMWebViewController.h [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+@end
+
+@protocol TMWebViewControllerDelegate <NSObject>
+@optional
+
+/*! @abstract Delegate callback called when the user taps the Done button. Upon this call, the view controller should be dismissed by caller. */
+- (void)webViewControllerDidFinish:(TMWebViewController *)controller;
+
+@end
```
Comment (Pearapps) [inline TMTumblrSDK/Authentication/TMWebViewController.h]: Can you add a newline here?

Comment (shpuntov) [inline TMTumblrSDK/Authentication/TMWebViewController.h]: Done

**Comment by Pearapps** [comment]
@shpuntov We will also need you to sign this form http://static.tumblr.com/zyubucd/GaTngbrpr/tumblr_corporate_contributor_license_agreement_v1__10-7-14.pdf and email to legal@tumblr.com so we can merge after all feedback is addressed!

thanks!

**Comment by shpuntov** [comment]
@Pearapps If a developer wants to steal user passwords, he can always show `UIWebView` to a user and load your OAuth page URL. I'm afraid removing `TMWebViewController` from this pod can't help us to protect Tumblr users.

**Comment by Pearapps** [comment]
@shpuntov Have you had a chance to send in the CLA form yet? Looking forward to getting this code in!

**Pull request merged**